cuqi/experimental/mcmc/_rto.py

```diff
@@ -272,15 +292,55 @@ def _choose_stepsize(self):
     def prior(self):
         return self.target.prior.gaussian
 
+    def compute_map_regularized(self):
+        """
+        Compute the MAP estimate of the regularized linear least squares problem.
+        The code is largely borrowed from step() and the only differences are that
+        1. the data vector y is not perturbed;
+        2. the initial point for the solver is set to the specified "initial_point",
+           instead of the current pooint.
+        """
+        y = self.b_tild
+
+        if self.solver == "FISTA":
+            sim = FISTA(self.M, y, self.proximal,
+                        self.initial_point, maxit = self.maxit, stepsize = self._stepsize, abstol = self.abstol, adaptive = self.adaptive)         
+        elif self.solver == "ADMM":
+            sim = ADMM(self.M, y, self.proximal,
+                        self.initial_point, self.penalty_parameter, maxit = self.maxit, inner_max_it = self.inner_max_it, adaptive = self.adaptive)
+        elif self.solver == "ScipyLinearLSQ":
+            A_op = sp.sparse.linalg.LinearOperator((sum([llh.distribution.dim for llh in self.likelihoods])+self.target.prior.dim, self.target.prior.dim),
+                                    matvec=lambda x: self.M(x, 1),
+                                    rmatvec=lambda x: self.M(x, 2)
+                                    )
+            sim = ScipyLinearLSQ(A_op, y, self.target.prior._box_bounds, 
+                                    max_iter = self.maxit,
+                                    lsmr_maxiter = self.inner_max_it, 
+                                    tol = self.abstol,
+                                    lsmr_tol = self.inner_abstol)
+        elif self.solver == "ScipyMinimizer":
+            # Adapt bounds format, as scipy.minimize requires a bounds format 
+            # different than that in scipy.lsq_linear.
+            bounds = [(self.target.prior._box_bounds[0][i], self.target.prior._box_bounds[1][i]) for i in range(self.target.prior.dim)]
+            # Note that the objective function is defined as 0.5*||Mx-y||^2, 
+            # and the corresponding gradient (gradfunc) is given by M^T(Mx-y).
+            sim = ScipyMinimizer(lambda x: 0.5*np.sum((self.M(x, 1)-y)**2), self.initial_point, gradfunc=lambda x: self.M(self.M(x, 1) - y, 2), bounds=bounds, tol=self.abstol, options={"maxiter": self.maxit})
+        else:
+            raise ValueError("Choice of solver not supported.")
+
+        self.map_regularized, _ = sim.solve()
```

Comment:
And this one as well, also we can call it `_map` for consistency with RTO
```suggestion
        self._map_regularized, _ = sim.solve()
```